packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -181,3 +306,54 @@ function isMixedLogicalExpression(node: TSESTree.LogicalExpression): boolean {
 
   return false;
 }
+
+function getOperator(
+  node: TSESTree.Expression,
+): '==' | '!=' | '===' | '!==' | undefined {
+  if (node.type === AST_NODE_TYPES.BinaryExpression) {
+    if (
+      node.operator === '==' ||
+      node.operator === '!=' ||
+      node.operator === '===' ||
+      node.operator === '!=='
+    ) {
+      return node.operator;
+    }
+  } else if (
+    node.type === AST_NODE_TYPES.LogicalExpression &&
+    node.left.type === AST_NODE_TYPES.BinaryExpression &&
+    node.right.type === AST_NODE_TYPES.BinaryExpression
+  ) {
+    if (node.operator === '||') {
+      if (node.left.operator === '===' && node.right.operator === '===') {
+        return '===';
+      } else if (
+        ((node.left.operator === '===' || node.right.operator === '===') &&
+          (node.left.operator === '==' || node.right.operator === '==')) ||
+        (node.left.operator === '==' && node.right.operator === '==')
+      ) {
+        return '==';
+      }
+    } else if (node.operator === '&&') {
+      if (node.left.operator === '!==' && node.right.operator === '!==') {
+        return '!==';
+      } else if (
+        ((node.left.operator === '!==' || node.right.operator === '!==') &&
+          (node.left.operator === '!=' || node.right.operator === '!=')) ||
+        (node.left.operator === '!=' && node.right.operator === '!=')
+      ) {
+        return '!=';
+      }
+    }
+  }
+  return;
+}
+
+function getNodes(node: TSESTree.Node): TSESTree.Node[] {
+  if (node.type === AST_NODE_TYPES.BinaryExpression) {
+    return [node.left, node.right];
+  } else if (node.type === AST_NODE_TYPES.LogicalExpression) {
+    return [...getNodes(node.left), ...getNodes(node.right)];
+  }
+  return [node];
```

Comment:
This line can't currently be reached, but I think it's still good to have here.

Reply:
> can't currently be reached

> good to have

Those two are incompatible 😛. 

Also, I'm a bit confused by this (rather vaguely named) function. Why retrieve all nodes here? Can you help me out please?

Reply:
If we have `a !== null && b !== undefined` it will give us `[a, null, b, undefined]`.
This is a pretty generic function, the reason it can't be reached is due to us only using it for one specific use case.

We only use it with either a binary or logical expression due to us using `getOperator` before, which fails if we use anything else.

Reply:
I will just inline `getOperator` and `getNodes`.